### PR TITLE
Add 'configure --enable-multilib': 32-/64-bit mode

### PR DESCRIPTION
**Pull request by gc00**
This PR depends on PR #784 being committed first.

This multilib mode is used by some customers of DMTCP.  By making it easier for all of us to generate multilib, we will be more likely to do testing for this important case.

This PR still has a bug in it, in the case when we restart a 32-bit application with `--enable-multilib`.

After doing `./configure --enable-multiib && make -j`, you can test this PR with:
```
make -j check-dmtcp1-32
```
which tests the 32-bit binary built from `dmtcp1.c`.

```
commit d586844a46c61fef3ec3d3bab646bfe31a65574c 
Date:   Sun Jul 21 10:33:18 2019 -0700
  SharedData: fixed header packing and member sizes. 
```

I have been able to isolate the problem only to this single commit on the 2.6 branch so far.  Something more complicated (but seemingly related) seems to be happening on the 3.0 branch.

I have reverted that commit, and seen that this also works for 32-bit mode.  (But the right fix will be to fix the commit d58684, of course, rather than revert it.)

As explained in configure.ac,
```
If --enable-multilib is invoked, then the top-level 'Makefile' will
   create config.status-multilib-m32.
  Invoking ./config.status-multilib-m32 configures as if --enable-m32 called.
  Invoking ./config.status configures as if --enable-m32 was not called.
```
PR #784 fixes the bug related to that commit.

The Makefile now automates the multilib procedure described in the DMTCP FAQ, for 'multi-arch/multilib' for how we build multilib manually.  It also provides the dmtcp1-32 test for testing a 32-bit application.

### CODE REVIEW

**Comment by gc00** [comment]
@karya0 @pgrosu , Can I get a review for this PR?  It's ready-to-go for 2.6.1, but I'd like to have a second pair of eyes examine it before I push it in.  It's fine to ask for changes in 3.0, since something subtle is not yet working there.

Essentially, you only need to look at `configure.ac` and `Makefile.in` (and a trivial one-character change in `plugin/Makefile.in`).  Everything else is auto-generated.

I'm asking for the 2.6.1-only review now, since we have customers who what the mixed 32-/64-bit mode.  And we should soon do a 2.6.1 release for aribitrary users who need that feature.  (And we can also announce for 2.6.1 the WSL/Windows version working for the single-process case.)

Thanks.

**Comment by karya0** [comment]
@gc00: https://github.com/dmtcp/dmtcp/pull/784 should take care of the DmtcpMessage size mismatch.

File: Makefile.in [hunk 1]
```diff
@@ -58,16 +59,64 @@ MANPAGES=dmtcp.1.gz dmtcp_coordinator.1.gz dmtcp_command.1.gz dmtcp_launch.1.gz
...
 	$(MKDIR_P) $(targetdir)/bin
 	$(MKDIR_P) $(targetdir)/lib/dmtcp
 
 build: mkdirs dmtcp plugin contrib
 ifeq ($(M32),1)
-	cd lib/dmtcp && $(LN_S) 32/lib/dmtcp/* ./
-	mkdir -p bin && cd bin && $(LN_S) ../lib/dmtcp/32/bin/* ./
+	cd lib/dmtcp && $(LN_S) 32/lib/dmtcp/* ./ || true
+	mkdir -p bin || true
+	cd bin && $(LN_S) ../lib/dmtcp/32/bin/* ./ || true
```
Comment (karya0) [inline Makefile.in]: Are the `|| true` statements intentional? Shouldn't we fail if we are unable to create the links? Perhaps you can add `-f` to force symlink creation?

File: configure.ac [hunk 2]
```diff
@@ -276,7 +276,7 @@ fi
 
 AC_ARG_ENABLE([m32],
             [AS_HELP_STRING([--enable-m32],
-                            [Compile in 32 bit mode on 64 bit linux. (Works on
+                            [Compile in 320bit mode on 64-bit Linux. (Works on
```
Comment (karya0) [inline configure.ac]: s/320/32-/

File: Makefile.in (hunk 1 above)
Comment (pgrosu) [reply Makefile.in]: I believe already that defined as such on the following line:

https://github.com/dmtcp/dmtcp/blob/bac3977fd0a49e4e95681c52bac0a8d01adb5ac4/Makefile.in#L2

File: configure [hunk 3]
```diff
@@ -1495,9 +1499,11 @@ Optional Features:
 
   --enable-pthread-mutex-wrappers
                           Recommended when using Open MPI
-  --enable-m32            Compile in 32 bit mode on 64 bit linux. (Works on
+  --enable-m32            Compile in 320bit mode on 64-bit Linux. (Works on
                           most, but not all programs, due to restrictions of
                           32-bit mode.)
+  --enable-multilib       Compile for mixed 320bit and 64-bit mode. (Requires
```
Comment (pgrosu) [inline configure]: s/320/32-

Comment (pgrosu) [inline configure]: s/320/32-

File: configure [hunk 4]
```diff
@@ -1495,9 +1499,11 @@ Optional Features:
 
   --enable-pthread-mutex-wrappers
                           Recommended when using Open MPI
-  --enable-m32            Compile in 32 bit mode on 64 bit linux. (Works on
+  --enable-m32            Compile in 320bit mode on 64-bit Linux. (Works on
```
Comment (pgrosu) [inline configure]: s/320/32-

File: configure [hunk 5]
```diff
@@ -2596,8 +2602,8 @@ test "$program_suffix" != NONE &&
 ac_script='s/[\\$]/&&/g;s/;s,x,x,$//'
 program_transform_name=`$as_echo "$program_transform_name" | sed "$ac_script"`
 
-# Expand $ac_aux_dir to an absolute path.
-am_aux_dir=`cd "$ac_aux_dir" && pwd`
+# expand $ac_aux_dir to an absolute path
```
Comment (pgrosu) [inline configure]: s/expand/Expand

I feel I'm getting to Google-level nit-picking, which I would like to avoid.

**Pull request merged**